File: core/src/com/jediterm/terminal/model/StyleState.java
package com.jediterm.terminal.model;

import com.jediterm.terminal.TerminalColor;
import com.jediterm.terminal.TextStyle;
import org.jetbrains.annotations.NotNull;

public class StyleState {
  private volatile TextStyle myCurrentStyle = TextStyle.EMPTY;
  private volatile TextStyle myDefaultStyle = TextStyle.EMPTY;

  public StyleState() {
  }

  public @NotNull TextStyle getCurrent() {
    return myCurrentStyle;
  }

  public void reset() {
    myCurrentStyle = myDefaultStyle;
  }

  public void set(StyleState styleState) {
    setCurrent(styleState.getCurrent());
  }

  public void setDefaultStyle(TextStyle defaultStyle) {
    myDefaultStyle = defaultStyle;
  }

  public TerminalColor getBackground() {
    return getBackground(null);
  }

  public TerminalColor getBackground(TerminalColor color) {
    return color != null ? color : myDefaultStyle.getBackground();
  }

  public TerminalColor getForeground() {
    return getForeground(null);
  }

  public TerminalColor getForeground(TerminalColor color) {
    return color != null ? color : myDefaultStyle.getForeground();
  }

  public void setCurrent(TextStyle current) {
    myCurrentStyle = current;
  }
}

File: bossterm-core-mpp/src/jvmMain/kotlin/ai/rever/bossterm/terminal/model/StyleState.kt
package ai.rever.bossterm.terminal.model

import ai.rever.bossterm.terminal.TextStyle
import java.util.*
import kotlin.concurrent.Volatile

class StyleState {
    @Volatile
    var current: TextStyle = TextStyle.Companion.EMPTY

    @Volatile
    private var myDefaultStyle: TextStyle = TextStyle.Companion.EMPTY

    fun reset() {
        this.current = myDefaultStyle
    }

    fun setDefaultStyle(defaultStyle: TextStyle) {
        myDefaultStyle = defaultStyle
    }

    val defaultBackground: ai.rever.bossterm.terminal.TerminalColor
        get() = Objects.requireNonNull<ai.rever.bossterm.terminal.TerminalColor?>(myDefaultStyle.background)

    val defaultForeground: ai.rever.bossterm.terminal.TerminalColor
        get() = Objects.requireNonNull<ai.rever.bossterm.terminal.TerminalColor?>(myDefaultStyle.foreground)
}
